PSReadLine/DynamicHelp.cs

```diff
@@ -0,0 +1,202 @@
+                        regexPatternToScrollTo = $"-[{upper}|{lower}]{remainingString} [<|\\[]";
+                    }
+
+                    _pager.Write(fullHelp, regexPatternToScrollTo);
+                }
+            }
+            else if (!String.IsNullOrEmpty(commandName) && !String.IsNullOrEmpty(parameterName))
+            {
+                _ps ??= System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace);
+                _ps.Commands.Clear();
+
+                PSObject paramHelp = _ps
+                    .AddCommand("Microsoft.PowerShell.Core\\Get-Help")
+                    .AddParameter("Name", commandName)
+                    .AddParameter("Parameter", parameterName)
+                    .Invoke<PSObject>()
+                    .FirstOrDefault();
+
+                WriteParameterHelp(paramHelp);
+            }
+        }
+
+        private void WriteDynamicHelpBlock(Collection<string> helpBlock)
+        {
+            var bufferWidth = _console.BufferWidth;
+            var colWidth = Math.Min(helpBlock.Max(s => LengthInBufferCells(s)) + 2, bufferWidth);
+            int columns = 1;
+
+            var dynHelp = new DynamicHelpBlock
+            {
+                Singleton = this,
+                ColumnWidth = colWidth,
+                Columns = columns,
+                Rows = (helpBlock.Count + columns - 1) / columns,
+                HelpItems = helpBlock
+            };
+
+            dynHelp.SaveCursor();
+            dynHelp.Draw(dynHelp);
+
+            Console.ReadKey(intercept: true);
+
+            dynHelp.Clear();
+            dynHelp.RestoreCursor();
+        }
+
+        private void WriteParameterHelp(dynamic helpContent)
+        {
+            Collection<string> helpBlock;
+
+            if (helpContent == null || String.IsNullOrEmpty(helpContent?.Description?[0]?.Text))
+            {
+                helpBlock = new Collection<string>()
+                {
+                    String.Empty,
+                    PSReadLineResources.NeedsUpdateHelp
+                };
+            }
+            else
+            {
+                char c = (char)0x1b;
+
+                string syntax = $"{c}[7m-{helpContent.name} <{helpContent.type.name}>{c}[0m";
+                string desc = "DESC: " + helpContent.Description[0].Text;
+                string details = $"Required: {helpContent.required}, Position: {helpContent.position}, Default Value: {helpContent.defaultValue}, Pipeline Input: {helpContent.pipelineInput}, WildCard: {helpContent.globbing}";
+
+                helpBlock = new Collection<string>
+                {
+                    String.Empty,
+                    syntax,
+                    String.Empty,
+                    desc,
+                    details
+                };
+            }
+
+            WriteDynamicHelpBlock(helpBlock);
+        }
+
+        private class DynamicHelpBlock : MultilineDisplayBlock
```

Comment:
We should have a more general purpose type to write out a collection of strings below the input buffer.

Reply:
Please elaborate.

Reply:
Fixed